aspnetcore/fundamentals/middleware/rate-limit/samples/WebRateLimitAuth/Program.cs

```diff
@@ -0,0 +1,197 @@
+#define ADMIN // FIRST ADMIN
+#if NEVER
+#elif FIRST
+// <snippet_1>
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.RateLimiting;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+using System.Threading.RateLimiting;
+using WebRateLimitAuth.Data;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var connectionString = builder.Configuration.GetConnectionString("DefaultConnection") ??
+    throw new InvalidOperationException("Connection string 'DefaultConnection' not found.");
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+    options.UseSqlServer(connectionString));
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+builder.Services.AddRazorPages();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseMigrationsEndPoint();
+}
+else
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseRouting();
+
+// <snippet>
+app.UseAuthentication();
+app.UseAuthorization();
+
+var userPolicyName = "user";
+
+var options = new RateLimiterOptions()
+    .AddPolicy<string>(userPolicyName, context =>
+    {
+        if (!context.User?.Identity?.IsAuthenticated ?? false)
```

Comment:
How is this supposed to work? For unauthenticated users the Identity.Name will be empty. And this check seems to skip null User/Identity entirely...

Reply:
If user is anonymous, they are rate limited under the `"anonymous user"` partition key. Yes, null user is under this partition.

Reply:
But the current code means that authenticated users and requests with no user info (unauthenticated) will end up using `CreateNoLimiter` partition, while unauthenticated users with empty user info will all end up using the same `CreateSlidingWindowLimiter` partition. What's the point of that?

Reply:
> But the current code means that authenticated users and requests with no user info (unauthenticated) will end up using `CreateNoLimiter` partition, while unauthenticated users with empty user info will all end up using the same `CreateSlidingWindowLimiter` partition. What's the point of that?

No, it's the opposite, that's what `!` means.

Reply:
@pentp oops, you're right. I forgot to change the  null-coalescing operator `?? ` from `false` to `true`. It's fixed now
`if (!context.User?.Identity?.IsAuthenticated ?? true)`